### ISSUE TITLE
Better compatibility with other mods.

### ISSUE THREAD

**Issue by MagmaSlime123** [mist475/MistLotrTweaks#1]
LOTR seems to be pretty incompatible with a lot of other random mods, which I thought I'd mention here in case you'd be willing to fix any of them.

General Issues:

- [ ] BiblioCraft and Storage Drawers Compatibility for LOTR's woods..

- [ ] Makes LOTR’s roads not have trees/caves/buildings generate on/close to them.
- [x] Fix many of LOTR’s blocks not having lang entries when Waila is active while looking at them.
- [ ] Allow editing trades via an in-game GUI for NPCs.
- [ ] Allow Serene Seasons’ seasons and weather-related features to function correctly in Middle-earth. (I think the mod crashes if used with LOTR by default. The mod should do what it normally does but to LOTR's plants/foliage.)
- [ ] With BetterHud, LOTR's bows and crossbows always have a 0 on their item, even if you have arrows or bolts on you.
- [ ] BowOverhaul's Bows and Quiver refuses to function. (Arrows in a Quiver don't allow shooting, arrows in your inventory aren't detected by its or LOTR's bows, etc.)
- [ ] FoodExpansion's Horse Meat dropping from most animals.

Compatibility Suggestions:

Et Futurum Requiem Compatibility
- [ ] Add boats, chest boats, signs, buttons, etc. and other missing wood-related items for LOTR wood.
- [ ] Make LOTR rabbits drop rabbit hide and feet. (Give feet a use of some kind. A new drink that gives a high level of jump boost, for example.)
- [ ] Raw Ores for LOTR's metals. (Copper exists in both mods, don't know how this would work.)
- [ ] Deepslate Ores for Middle-earth ores that generate near/close to bedrock. (Would cancel their normal versions from generating.)
- [ ] Make Amethyst Geodes generate, stop LOTR Amethyst veins from generating, and alter trades/recipes that use LOTR Amethyst to use Et Futurum Amethyst Shards instead.
- [x] Allow Bee Nests and Bees to generate.
- [ ] Allow Cornflowers and Lilly of the Valley flowers to generate.
- [ ] Composter support for LOTR's plants.
- [x] Bark blocks made of LOTR's wood. (implemented by Jonosa in [METweaks](https://modrinth.com/mod/middleearth-tweaks))

Exotic Birds Compatibility
- [ ] Make Exotic Birds’ birds spawn in their respective Middle-earth biomes.
- [ ] Make Exotic Birds’ birds support LOTR bird cages.
- [ ] Remove LOTR bird mobs from spawning.
- [ ] Remove Unidentified Eggs from bird nests.
- [ ] Remove Phoenixs and any loot/eggs/etc. related to them.
- [ ] Remove any birds that wouldn’t fit in LOTR.

LOTR Aquaculture Support
- [ ] Adds in-world versions of Aquaculture’s fish and makes them spawn in their respective biomes (both work together in Renewed).
- [ ] Adds trades for Aquaculture’s items.

Dynamic Trees Compatibility
- [ ] Replace normal LOTR trees with Dynamic Trees versions.

LOTR Wild Mobs Compatibility
- [ ] Make Wild Mobs mobs spawn in their respective ME biomes
- [ ] Remove Middle-earth Butterflies and replace them with Wild Mobs’ versions, or allow LOTR’s Butterflies to be caught using
Wild Mobs’ nets. If released, return them to LOTR Butterflies or Wild Mobs Butterflies (depending on what you do).

**Comment by mist475** [comment]
This is very much a project I work on when I feel like it. When I work on it again I'll keep the list in mind though the things on there might not have the highest priority.

**Comment by MagmaSlime123** [comment]
> This is very much a project I work on when I feel like it. When I work on it again I'll keep the list in mind though the things on there might not have the highest priority.

Ah, alright. Thought I'd mention some of these issues since LOTR lacks compatibility with some other mods.

**Comment by ArchiDreamZ** [comment]
I think the compatibility of Lord of the Rings armor textures with Mo' Bends is the highest priority, which can maximize the player experience.

**Comment by DarianLStephens** [comment]
I really just want to be able to control generation in LoTR biomes.
Being able to disable more ores or add new ones that will generate only in specified biomes would be _great_ for a modpack with more travel focus!
The redstone config here is a perfect example of that.

**Comment by MagmaSlime123** [comment]
> I really just want to be able to control generation in LoTR biomes. Being able to disable more ores or add new ones that will generate only in specified biomes would be _great_ for a modpack with more travel focus! The redstone config here is a perfect example of that.

Agreed. I also have a modpack that uses Et Futurum's Amethyst Geodes, and thus makes LOTR's Amethyst Veins useless.

If you want to chat about this pack on my Discord, feel free to join, https://discord.gg/TBu4QBmSzf.

**Comment by mist475** [comment]
> LOTR seems to be pretty incompatible with a lot of other random mods, which I thought I'd mention here in case you'd be willing to fix any of them.
> 
> General Issues:
> 
>     * BiblioCraft and Storage Drawers Compatibility for LOTR's woods..
> 
>     * Makes LOTR’s roads not have trees/caves/buildings generate on/close to them.
> 
>     * Fix many of LOTR’s blocks not having lang entries when Waila is active while looking at them.
> 
>     * Allow editing trades via an in-game GUI for NPCs.
> 
>     * Allow Serene Seasons’ seasons and weather-related features to function correctly in Middle-earth. (I think the mod crashes if used with LOTR by default. The mod should do what it normally does but to LOTR's plants/foliage.)
> 
>     * With BetterHud, LOTR's bows and crossbows always have a 0 on their item, even if you have arrows or bolts on you.
> 
>     * BowOverhaul's Bows and Quiver refuses to function. (Arrows in a Quiver don't allow shooting, arrows in your inventory aren't detected by its or LOTR's bows, etc.)
> 
>     * FoodExpansion's Horse Meat dropping from most animals.
> 
> 
> Compatibility Suggestions:
> 
> Et Futurum Requiem Compatibility
> 
>     * Add boats, chest boats, signs, buttons, etc. and other missing wood-related items for LOTR wood.
> 
>     * Make LOTR rabbits drop rabbit hide and feet. (Give feet a use of some kind. A new drink that gives a high level of jump boost, for example.)
> 
>     * Raw Ores for LOTR's metals. (Copper exists in both mods, don't know how this would work.)
> 
>     * Deepslate Ores for Middle-earth ores that generate near/close to bedrock. (Would cancel their normal versions from generating.)
> 
>     * Make Amethyst Geodes generate, stop LOTR Amethyst veins from generating, and alter trades/recipes that use LOTR Amethyst to use Et Futurum Amethyst Shards instead.
> 
>     * Allow Bee Nests and Bees to generate.
> 
>     * Allow Cornflowers and Lilly of the Valley flowers to generate.
> 
>     * Composter support for LOTR's plants.
> 
>     * Bark blocks made of LOTR's wood.
> 
> 
> Exotic Birds Compatibility
> 
>     * Make Exotic Birds’ birds spawn in their respective Middle-earth biomes.
> 
>     * Make Exotic Birds’ birds support LOTR bird cages.
> 
>     * Remove LOTR bird mobs from spawning.
> 
>     * Remove Unidentified Eggs from bird nests.
> 
>     * Remove Phoenixs and any loot/eggs/etc. related to them.
> 
>     * Remove any birds that wouldn’t fit in LOTR.
> 
> 
> LOTR Aquaculture Support
> 
>     * Adds in-world versions of Aquaculture’s fish and makes them spawn in their respective biomes (both work together in Renewed).
> 
>     * Adds trades for Aquaculture’s items.
> 
> 
> Dynamic Trees Compatibility
> 
>     * Replace normal LOTR trees with Dynamic Trees versions.
> 
> 
> LOTR Wild Mobs Compatibility
> 
>     * Make Wild Mobs mobs spawn in their respective ME biomes
> 
>     * Remove Middle-earth Butterflies and replace them with Wild Mobs’ versions, or allow LOTR’s Butterflies to be caught using
>       Wild Mobs’ nets. If released, return them to LOTR Butterflies or Wild Mobs Butterflies (depending on what you do).

As I've been using waila and Wisla (lotr waila addon) recently. I've fixed the missing names issue with a pr to Wisla which got released as this update: https://legacy.curseforge.com/minecraft/mc-mods/wisla/files/5371214

**Comment by MagmaSlime123** [comment]
As of the past few months some of these aren't exactly needed. Et Futurum's compatibility has gotten much better, but some stuff related to worldgen, wood variants, and a few other things still stands.

**Comment by mist475** [comment]
I added configurable efr beehive generation to middle earth in the latest update, enjoy your beekeeping

**Comment by Novembeetle** [comment]
Could you make chandeliers be able to hang from chains and end rods from Et Futurum please?
